filesystems/watcher.go

```diff
@@ -68,24 +68,26 @@ var defaultUploadRegexpList = []*regexp.Regexp{
 }
 
 type WatchConfig struct {
-	localDir              string
-	uploadDir             string
-	uploadRegexp          *regexp.Regexp
-	uploadCheckInterval   time.Duration // The interval at which we check for files that need uploading.
-	uploadThresholdTime   time.Duration // Duration for which a file must be untouched to be uploader.
-	stdioLogCheckInterval time.Duration
-	keepFileAfterUpload   bool
+	localDir               string
+	uploadDir              string
+	uploadRegexp           *regexp.Regexp
+	uploadCheckInterval    time.Duration // The interval at which we check for files that need uploading.
+	uploadThresholdTime    time.Duration // Duration for which a file must be untouched to be uploader.
+	stdioLogCheckInterval  time.Duration
+	keepFileAfterUpload    bool
+	finalUploadMatchRegexp bool
 }
 
-func NewWatchConfig(localDir, uploadDir string, uploadRegexp *regexp.Regexp, uploadCheckInterval, uploadThresholdTime, stdioLogCheckInterval time.Duration, keepFileAfterUpload bool) WatchConfig {
+func NewWatchConfig(localDir, uploadDir string, uploadRegexp *regexp.Regexp, uploadCheckInterval, uploadThresholdTime, stdioLogCheckInterval time.Duration, keepFileAfterUpload, finalUploadMatchRegexp bool) WatchConfig {
```

Comment:
I don't think we need this layer of indirection. It adds no value as a constructor hides the keyword/order of arguments in the calling function.